### DISCUSSION TITLE
Lesson 13 eth20faucet

### DISCUSSION THREAD

**Discussion by sk67dev** [smartcontractkit/full-blockchain-solidity-course-py#276] [Q&A]
Hi,

The 'Mint Free Tokens' Button isn't working for erc20faucet.com. I connected with MetaMask; My address was entered and I tried with different amounts but each time nothing happens when I tap the button. Am I doing something wrong or is there an alternative I can use please?

**Comment by cromewar** [comment]
Yes, you are right is not working I've tried with different networks, however you can use a more rustic alternative:

[WeenusTokenFaucet](https://github.com/bokkypoobah/WeenusTokenFaucet)

As the documentation says:

- Search in your Ethereum blockchain explorer for WEENUS (18 decimals), XEENUS (18 decimals), YEENUS (8 decimals) or ZEENUS (0 decimals).

- Send a 0 value transaction from your account to the token contract addresses below on your preferred network, and these token contracts will mint 1,000 ERC20 test tokens for your account.

For example for Rinkeby you should send a Zero transaction to this address: 0xaFF4481D10270F50f203E0763e2597776068CBc5 you can check it on the documentation above




![image](https://user-images.githubusercontent.com/20199180/139431321-981aaad7-6dde-4b80-bbde-c7b17bf14d6f.png)


![image](https://user-images.githubusercontent.com/20199180/139431713-e415674e-de79-4895-a897-f4fdbc3027b9.png)

After adding the same contract to metamask you'll see you now have 1000 Wennus ERC20 token

![image](https://user-images.githubusercontent.com/20199180/139431872-b8eed4b0-ba5d-4b61-8d20-bd76910409af.png)
